code-challenges.test.js

```diff
@@ -21,47 +23,86 @@ var secretCodeWord1 = "lackadaisical"
 // Expected output: "l4ck4d41s1c4l"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
-
-
-
-
-
-// b) Create the function that makes the test pass.
-
-
-
-
-
+describe('when codedMessage is called', () => {
+    it ('returns l4ck4d41s1c4l', ()=> {
+        let secretCodeWord1 = "lackadaisical"
+        let wordCoded = codedMessage(secretCodeWord1)
+        expect(wordCoded).toEqual("l4ck4d41s1c4l")
+    })
+    it ('returns g0bbl3dyg00k', () => {
+        let secretCodeWord2 = "gobbledygook"
+        let wordCoded2 = codedMessage(secretCodeWord2)
+        expect(wordCoded2).toEqual("g0bbl3dyg00k")
+    })
+})
+// // b) Create the function that makes the test pass.
+const codedMessage = (strr) => {
+    let replacements = ['4', '3', '1', 0]
+    return strr.split('').map(value => {
+      return (value === 'a') ? replacements[0] : 
```

Comment:
ternary operators are good for short concise logic, not for logic that is nested which this is